lib/pbench/test/unit/agent/conftest.py

```diff
@@ -39,39 +40,19 @@ def do_setup(tmp_path_factory):
     with open(cfg_file, "w") as fp:
         fp.write(agent_cfg_tmpl.format(TMP=tmp_d))
 
-    return dict(tmp=tmp_d, cfg_dir=pbench_cfg)
-
-
-def on_disk_agent_config(tmp_path_factory):
-    """Base setup function shared between the agent and functional tests.
-
-    See https://github.com/pytest-dev/pytest-xdist.
-    """
-    root_tmp_dir = tmp_path_factory.getbasetemp()
-    marker = root_tmp_dir / "agent-marker.json"
-    with FileLock(f"{marker}.lock"):
-        if marker.is_file():
-            the_setup = json.loads(marker.read_text())
-            the_setup["tmp"] = Path(the_setup["tmp"])
-            the_setup["cfg_dir"] = Path(the_setup["cfg_dir"])
-        else:
-            the_setup = do_setup(tmp_path_factory)
-            data = dict(tmp=str(the_setup["tmp"]), cfg_dir=str(the_setup["cfg_dir"]))
-            marker.write_text(json.dumps(data))
-    return the_setup
+    return pbench_cfg
 
 
 @pytest.fixture(scope="session", autouse=True)
 def agent_setup(tmp_path_factory):
     """Test package setup for agent unit tests"""
-    on_disk_agent_config(tmp_path_factory)
+    return on_disk_config(tmp_path_factory, "agent", do_setup)
 
 
 @pytest.fixture(autouse=True)
-def setup(tmp_path_factory, monkeypatch):
+def setup(tmp_path_factory, agent_setup, monkeypatch):
```

Comment:
Just an observation:  since `setup()` is an auto-use fixture which references the `agent_setup()` fixture, that means that`agent_setup()` will always be used, regardless of whether it is marked as `autouse` or not.